assume/strategies/flexable_storage.py

```diff
@@ -33,54 +34,186 @@ def calculate_bids(
+            )
+
+        energy_price = capacity_price / unit.current_SOC
+
+        if market_config.product_type == "capacity_pos":
+            bids = [
+                {"price": capacity_price, "volume": bid_quantity},
+            ]
+        elif market_config.product_type == "energy_pos":
+            bids = [
+                {"price": energy_price, "volume": bid_quantity},
+            ]
+        else:
+            raise ValueError(
+                f"Product {market_config.product_type} is not supported by this strategy."
+            )
+
+        return bids
 
 
-class flexableCRMStorage(BaseStrategy):
+class flexableNegCRMStorage(BaseStrategy):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
-        self.foresight = pd.Timedelta("12h")
+        self.foresight = pd.Timedelta(kwargs.get("crm_foresight", "4h"))
+
+        self.current_time = None
 
     def calculate_bids(
         self,
         unit: BaseUnit,
         operational_window: OperationalWindow,
         market_config: MarketConfig,
+        data_dict: dict,
         **kwargs,
     ):
-        pass
+        # in flexable no prices calculated for CRM_neg
+        bid_quantity = operational_window["ops"]["neg_reserve"]["volume"]
+        if bid_quantity == 0:
+            return []
+        # if bid_quantity >= min_bid_volume
+
+        if market_config.product_type == "capacity_neg":
+            bids = [
+                {"price": 0, "volume": bid_quantity},
+            ]
+        elif market_config.product_type == "energy_neg":
+            bids = [
+                {"price": 0, "volume": bid_quantity},
+            ]
+        else:
+            raise ValueError(
+                f"Product {market_config.product_type} is not supported by this strategy."
+            )
+
+        return bids
+
+
+def calculate_price_average(unit, current_time, foresight, price_forecast):
+    average_price = np.mean(
+        price_forecast[current_time - foresight : current_time + foresight]
+    )
+
+    return average_price
+
+
+def get_specific_revenue(unit, marginal_cost, current_time, foresight, price_forecast):
+    t = current_time
+
+    if t + foresight > price_forecast.index[-1]:
+        price_forecast = price_forecast.loc[t:]
+    else:
+        price_forecast = price_forecast.loc[t : t + foresight]
+
+    possible_revenue = 0
+    theoretic_SOC = unit.current_SOC
+    for market_price in price_forecast:
+        theoretic_power_discharge = min(
+            max(theoretic_SOC - unit.min_SOC, 0), unit.max_power_discharge
+        )
+        possible_revenue += (market_price - marginal_cost) * theoretic_power_discharge
+        theoretic_SOC -= theoretic_power_discharge
+
+    if unit.current_SOC - theoretic_SOC != 0:
+        possible_revenue = possible_revenue / (unit.current_SOC - theoretic_SOC)
+
+    return possible_revenue
+
+    """
+    def calculatingBidPricesSTO_CRM(self, t):
```

Comment:
should we keep this here?

Reply:
No, it was just to have a comparison to flexABLE